core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -1424,33 +1423,47 @@ protected void bitcoinSerializeToStream(OutputStream stream) throws IOException
     }
 
     /**
-     * Serialize according to <a href="https://github.com/bitcoin/bips/blob/master/bip-0144.mediawiki">BIP144</a> or the
-     * <a href="https://en.bitcoin.it/wiki/Protocol_documentation#tx">classic format</a>, depending on if segwit is
-     * desired.
+     * Write this transaction into the given buffer.
+     *
+     * @param buf buffer to write into
+     * @return the buffer
+     * @throws BufferOverflowException if the transaction doesn't fit the remaining buffer
      */
-    protected void bitcoinSerializeToStream(OutputStream stream, boolean useSegwit) throws IOException {
+    protected ByteBuffer write(ByteBuffer buf, boolean useSegwit) throws BufferOverflowException {
```

Comment:
I think this can (and should) be made `private`

Reply:
It's protected, because it's overridden in `HugeDeclaredSizeTransaction` in a similar fashion that the previous `.bitcoinSerializeToStream`.